test/e2e/catalog_e2e_test.go

```diff
@@ -1069,26 +1068,33 @@ var _ = Describe("Catalog represents a store of bundles which OLM can use to ins
 		}
 
 		By("creating a catalog source")
-		source, err := crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Create(context.TODO(), source, metav1.CreateOptions{})
-		Expect(err).ToNot(HaveOccurred())
-
-		By("update the catalog source with template annotation")
 
-		source, err = crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Get(context.TODO(), source.GetName(), metav1.GetOptions{})
-		Expect(err).ShouldNot(HaveOccurred(), "error getting catalog source")
+		var err error
+		Eventually(func() error {
+			source, err = crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Create(context.Background(), source, metav1.CreateOptions{})
+			return err
+		}).Should(Succeed())
 
-		// create an annotation using the kube templates
-		source.SetAnnotations(map[string]string{catalogsource.CatalogImageTemplateAnnotation: fmt.Sprintf("quay.io/olmtest/catsrc-update-test:%s.%s.%s", catalogsource.TemplKubeMajorV, catalogsource.TemplKubeMinorV, catalogsource.TemplKubePatchV)})
+		By("updating the catalog source with template annotation")
 
-		// Update the catalog image
 		Eventually(func() error {
-			source, err = crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Update(context.TODO(), source, metav1.UpdateOptions{})
+			source, err = crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Get(context.Background(), source.GetName(), metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			// create an annotation using the kube templates
+			source.SetAnnotations(map[string]string{
+				catalogsource.CatalogImageTemplateAnnotation: fmt.Sprintf("quay.io/olmtest/catsrc-update-test:%s.%s.%s", catalogsource.TemplKubeMajorV, catalogsource.TemplKubeMinorV, catalogsource.TemplKubePatchV),
+			})
```

Comment:
nit: we _could_ define the map literal outside of this closure and set it here too